File: G-for-g-Practice/Second largest.py
#User function Template for python3
class Solution:
    def getSecondLargest(self, arr):
        if len(arr)<2:
            return -1
        big, bigger = float('-inf'),float('-inf')
        
        for num in arr:
            if num>bigger:
                big= bigger
                bigger = num
            elif bigger>num> big:
                big= num
        
        return big if big!=float('-inf') else -1

File: G-for-g-Practice/Second largest.java




// User function Template for Java

class Solution {
    public int getSecondLargest(int[] arr) {
        // Code Here
        if (arr.length<2) return -1;
        
        int big = Integer.MIN_VALUE, bigger = Integer.MIN_VALUE;
        
        for (int num: arr){
            if (num>bigger){
                big= bigger;
                bigger= num;
            }
            else if ( big <num && num < bigger){
                big= num;
            }
        }
        return (big!= Integer.MIN_VALUE)?big: -1;
    }
}
